components/app/App.js

```diff
@@ -0,0 +1,77 @@
+import Keyboard from '../keyboard/Keyboard.js';
+import KeyboardTemplate from '../../data/BUTTONS.js';
+
+export default class App {
+  constructor() {
+    this._textArea = null;
+
+    this._keyboard = null;
+
+    this._setListeners();
+  }
+
+  buildApplication() {
+    this._createTextArea();
+    this._keyboard = new Keyboard(document.body, KeyboardTemplate);
+  }
+
+  _createTextArea() {
+    const textArea = document.createElement('textarea');
+
+    textArea.classList.add('textarea');
+    textArea.cols = 70;
+    textArea.rows = 10;
+    document.body.appendChild(textArea);
+
+    this._textArea = textArea;
+  }
+
+  _setListeners() {
+    let lastKey = '';
+    document.addEventListener('keydown', (event) => {
+      event.preventDefault();
+
+
+      const button = document.querySelector(`[data-code = ${event.code}]`);
+
+      if (button) {
+        if ((lastKey === 'Control' && event.key === 'Alt') || (lastKey === 'Alt' && event.key === 'Control')) this._keyboard.switchLanguage();
+        lastKey = event.key;
+
+        const functionalKeys = {
+
+          'control-keys': {
+            Control: '',
+            Alt: '',
+            Meta: '',
+            Tab: ' '.repeat(10),
+            Enter: '\n',
+          },
+
+          'upper-case-keys': ['CapsLock', 'Shift'],
+        };
```

Comment:
I would take it out from here and declare as config in a separate file or just declare it outside the event listener callback.